### PR TITLE
workspace: Improve error handling when dropping a file that cannot be opened into the workspace pane

### PR DESCRIPTION
**Pull request by ssut**
This PR can improve the UX when dropping a file that cannot be opened into the workspace pane. Previously, nothing happened without any messages when such error occurred, which could be awkward for users. Additionally the pane was being split even though the file failed to open.

Here's a screen recording demonstrating the previous/updated behavior:

https://github.com/user-attachments/assets/cfdf3488-9464-4568-b16a-9b87718bd729

Changes:

- It now displays an error message if a file cannot be opened.
- Updated the logic to first try to open the file. The pane splits only if the file opening process is successful.

Release Notes:

- Improved error handling when opening files in the workspace pane. An error message will now be displayed if the file cannot be opened.
- Fixed an issue where unnecessary pane splitting occurred when a file fails to open.

### CODE REVIEW

**Comment by mrnugget** [comment]
Hey! I'm pretty sure you can simplify your code by using this method:

https://github.com/zed-industries/zed/blob/3f3ed89b63520964b4e653d56c8d3539f549e867/crates/workspace/src/notifications.rs#L585-L595

You can see how it's used here:

https://github.com/zed-industries/zed/blob/3f3ed89b63520964b4e653d56c8d3539f549e867/crates/terminal_view/src/terminal_view.rs#L134-L139

**Comment by ssut** [comment]
> Hey! I'm pretty sure you can simplify your code by using this method:
> 
> https://github.com/zed-industries/zed/blob/3f3ed89b63520964b4e653d56c8d3539f549e867/crates/workspace/src/notifications.rs#L585-L595
> 
> You can see how it's used here:
> 
> https://github.com/zed-industries/zed/blob/3f3ed89b63520964b4e653d56c8d3539f549e867/crates/terminal_view/src/terminal_view.rs#L134-L139

Thanks for letting me know, I refactored the code accordingly calling `notify_async_err`.

File: crates/workspace/src/pane.rs [hunk 1]
```diff
@@ -2017,13 +2018,30 @@ impl Pane {
...
-                        workspace
-                            .open_path(path, Some(to_pane.downgrade()), true, cx)
-                            .detach_and_log_err(cx);
-                    }
+                        let load_path_task = workspace.load_path(path, cx);
+                        cx.spawn(|workspace, mut cx| async move {
+                            if let Some((project_entry_id, build_item)) =
+                                load_path_task.await.notify_async_err(&mut cx)
+                            {
+                                _ = workspace.update(&mut cx, |workspace, cx| {
```
Comment (mrnugget) [inline crates/workspace/src/pane.rs:2026]: I think the `_ =` can also be replaced with a `.log_err()` instead. Same as below.

**Review by mrnugget — APPROVED** [review]

**Comment by mrnugget** [comment]
Works for me! Thanks!
https://github.com/user-attachments/assets/dcd959df-c49b-48f7-a149-5e0f9db97c4a

**Pull request merged**